src/main/java/org/xstefank/model/PersistentList.java

```diff
@@ -0,0 +1,67 @@
+package org.xstefank.model;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PersistentList {
+
+    private List<String> list = new ArrayList<>();
+    private File file;
+
+    public PersistentList(String dirName, String fileName) {
+        if (dirName == null || fileName == null) {
+            throw new IllegalArgumentException("Parameters in " + this.getClass().getName() + " cannot be null");
+        }
+        file = new File(dirName, fileName);
+        if (!file.exists()) {
+            try {
+                file.createNewFile();
+            } catch (IOException e) {
+                throw new IllegalStateException("Cannot create new file with name " + file.getName(), e);
+            }
+        } else {
+            loadFileContent(file);
+        }
+    }
+
+    public PersistentList(File file) {
+        this.file = file;
+        if (file.exists()) {
+            loadFileContent(file);
+        }
```

Comment:
I can call "createNewFile" method but the FileWriter I am using in "addUser" method, automatically creates new file if it hasn't existed when is called. It is working as you would expect. The method "hasUsername" can be called without created file because that file is not used there.

Reply:
so what is the point of other constructor creating file if it does not exist?